larpandoracontent/LArHelpers/LArMCParticleHelper.cc

```diff
@@ -604,4 +640,37 @@ CaloHitList LArMCParticleHelper::GetSharedHits(const CaloHitList &hitListA, cons
     return sharedHits;
 }
 
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+MCParticleWeightMap LArMCParticleHelper::GetMCParticleWeightMap(const CaloHitList &caloHitList)
+{
+    MCParticleWeightMap mcParticleWeightMap;
+
+    for (const CaloHit *const pCaloHit : caloHitList)
+    {
+        // Synthesize the weights from every hit into the main map
+        const MCParticleWeightMap &hitMCWeightMap(pCaloHit->GetMCParticleWeightMap());
+        MCParticleVector           mcParticleVector;
+
+        // Iterate through map in a reproducible order
+        for (const MCParticleWeightMap::value_type &mapEntry : hitMCWeightMap)
+            mcParticleVector.push_back(mapEntry.first);
+
+        std::sort(mcParticleVector.begin(), mcParticleVector.end(), PointerLessThan<MCParticle>());
+
+        for (const MCParticle *const pMCParticle : mcParticleVector)
+        {
+            const auto findIter = mcParticleWeightMap.find(pMCParticle);
```

Comment:
Our use of auto is still a little inconsistent. This is indeed OK, but consistent use (or selective use, or even disuse) is something to think about in general.